### PR TITLE
add metadata validation

### PR DESCRIPTION
**Pull request by Katinka666**
related #22

### CODE REVIEW

**Review by borditamas — CHANGES_REQUESTED** [review]
ServiceDiscovery has been already merged into the development, please merge the development into your branch and add this validation also there. Do not forget also the interface properties

**Comment by Katinka666** [comment]
@borditamas Please review again

File: serviceregistry/src/main/java/eu/arrowhead/serviceregistry/service/validation/ServiceDiscoveryValidation.java [hunk 1]
```diff
@@ -107,6 +108,10 @@ public void validateRegisterService(final ServiceInstanceRequestDTO dto, final S
 				throw new InvalidParameterException("Interface properties are missing", origin);
 			}
 		}
+		
+		if (!Utilities.isEmpty(dto.metadata())) {
```
Comment (borditamas) [inline serviceregistry/src/main/java/eu/arrowhead/serviceregistry/service/validation/ServiceDiscoveryValidation.java]: Would be better to do this before the interface related parts.

File: serviceregistry/src/main/java/eu/arrowhead/serviceregistry/service/validation/ServiceDiscoveryValidation.java [hunk 2]
```diff
@@ -107,6 +108,10 @@ public void validateRegisterService(final ServiceInstanceRequestDTO dto, final S
 				throw new InvalidParameterException("Interface properties are missing", origin);
 			}
```
Comment (borditamas) [inline serviceregistry/src/main/java/eu/arrowhead/serviceregistry/service/validation/ServiceDiscoveryValidation.java:115]: Here we shoud also validate the keys of the interface properties (which are basically metadata as well)
`MetadataValidation.validateMetadataKey(interfaceDTO.properties());`

**Review by borditamas — CHANGES_REQUESTED** [review]

**Review by rbocsi — APPROVED** [review]

**Review by borditamas — APPROVED** [review]

**Pull request merged**